march_hardware/src/ethercat/ethercat_master.cpp

```diff
@@ -95,9 +95,9 @@ bool EthercatMaster::ethercatSlaveInitiation(std::vector<Joint>& joints)
 
   for (Joint& joint : joints)
   {
-    if (joint.hasIMotionCube())
+    if (joint.getMotorControllerSlaveIndex() > -1)
```

Comment:
```suggestion
    if (joint.getMotorControllerSlaveIndex() <= 0)
```

Reply:
No? This should only happen when for ethercat motor controllers, so any motor controller with a slave index > -1. (-1 means no motor controller, and I think Imma use -2 for a non-ethercat motor controller. Or just -1 as well)

Reply:
Sorry I meant greater or equal then zero, thought it was a bit more intuitive. (but it is a minor detail, so don't bother to much).